game/src/core/hud/DialogDesign.java

```diff
@@ -82,7 +82,8 @@ private void VisualizeQuestionSection(
         switch (questionContentType) {
             case TEXT -> addActor(createScrollPane(skin, new Label(outputMsg, skin)));
             case IMAGE -> addActor(
-                    createScrollPane(skin, new Image(new Texture(ImagePathExtractor(outputMsg).get()))));
+                    createScrollPane(
+                            skin, new Image(new Texture(ImagePathExtractor(outputMsg).get()))));
```

Comment:
Sorry für den Kommentar von der Seitenlinie: Wenn ich einen `ImagePathExtractor` habe, warum muss ich daraus erst eine `Texture` machen, die ich dann in ein `Image` stecke? Wäre es denkbar, sowas direkter zu machen, etwa: `new Image(ImagePathExtractor(outputMsg))`?

Reply:
> Sorry für den Kommentar von der Seitenlinie: Wenn ich einen `ImagePathExtractor` habe, warum muss ich daraus erst eine `Texture` machen, die ich dann in ein `Image` stecke? Wäre es denkbar, sowas direkter zu machen, etwa: `new Image(ImagePathExtractor(outputMsg))`?

Finde den Input wichtig, da ich sehr viel umgebaut habe. Für mich sah es sinnvoll aus, dieses nach außen zu geben als Optional. Da aktuell noch kein Kontrollieren von 
 - existiert ein Pfad im "output" String (regex findet nichts)
 - existiert eine Datei an diesem Pfad (File(Pfad) nicht erreichbar)

Aber beim erneuten Überlegen ist es vielleicht schlau, diese auch noch schlecht benannte Methode vielleicht doch nur einen String wiedergeben lassen.


> Edit: D.h. hier müssten evtl. noch passende Schnittstellen bereitgestellt werden?

Naja wir haben den Texturehandler ja entfernt. Benötigt werden würde eine `Texturehandler.get(stringPath: String) : libgdx.Texture` welche im `libgdx.scene2d` benutzt wird. Wir könnten für ein nächstes Issue überlegen für alle scene2d elemente eine Factory zu basteln welche umgeht das direkt mit libgdx im ui gearbeitet werden muss.


> Edit: Das `get()` könnte/sollte auch das `Image` mit erledigen, das ist hier unnötiger Ballast.

Solange es nicht extra kontrolliert wird, ja. Ich hatte es zum Testen benutzt, um herauszufinden, welche Quiz Demos falsch konfiguriert sind. Werde es glaube ich aber entfernen. Da ich aktuell immer, wenn ich was im Hinterkopf habe versuche es von außen konfigurierbar zu machen, ohne den Quellcode zu ändern. Bin ich mir nicht sicher, ob ein `Function<String, String>` für das Suchen von dem Bildpfad anbietet.

Reply:
@AHeinisch An welcher Stelle hattest Du Dir `Function<String, String>` vorgestellt? Als Parameter für `Image`? Dann würde ich als Nutzer den String nicht selbst ausrechnen, sondern die Berechnungsvorschrift übergeben, und es könnte für eine Handvoll von Use-Cases vordefinierte Funktionen geben? Hmmm 🤔

Reply:
@AHeinisch Hmmm, Du hättest auch einfach sagen können, `Image` ist (genau wie `Texture`) eine libGDX-Klasse. Dann hätte sich mein ganzer Kommentar direkt erledigt - da können wir ja nicht rein ...

Reply:
> Ich würde nur die restlichen Zwischenschritte eher in das `Image` schieben, d.h. der Ctor von `Image` müsste dann das `.get()` auf dem Optional aufrufen und ggf. daraus noch eine `Texture` bauen. Damit hätte ich als "Kunde" der API nur den `ImagePathExtractor(outputMsg)` und dann das neue `Image` und müsste mich mit den Zwischenschritten nicht herumplagen.

Leider benötigen wir für die libgdx UI Elemente direkt die Textur, welche ggfls. durch `TextureMap.getInstance().getTexture(texturePath)` ersetzt werden kann. Dies würde theoretisch identisch sein zu `new Textur(texturePath)`.

> @AHeinisch An welcher Stelle hattest Du Dir `Function<String, String>` vorgestellt? Als Parameter für `Image`? Dann würde ich als Nutzer den String nicht selbst ausrechnen, sondern die Berechnungsvorschrift übergeben, und es könnte für eine Handvoll von Use-Cases vordefinierte Funktionen geben? Hmmm 🤔

Also ich hatte es mir gedacht, um das Extrahieren des Dateipfades aus der Aufgabenstellung von außen konfigurierbar zu machen, anstelle eines fest codierten regex. Dies ist aber glaube eher was für ein neues Issue.

Reply:
> > Ich würde nur die restlichen Zwischenschritte eher in das `Image` schieben, d.h. der Ctor von `Image` müsste dann das `.get()` auf dem Optional aufrufen und ggf. daraus noch eine `Texture` bauen. Damit hätte ich als "Kunde" der API nur den `ImagePathExtractor(outputMsg)` und dann das neue `Image` und müsste mich mit den Zwischenschritten nicht herumplagen.
> 
> Leider benötigen wir für die libgdx UI Elemente direkt die Textur, welche ggfls. durch `TextureMap.getInstance().getTexture(texturePath)` ersetzt werden kann. Dies würde theoretisch identisch sein zu `new Textur(texturePath)`.

Hmmm, das ist Mist, das haben wir beim "Aufräumen" des `TextureHandler` übersehen.

@AMatutat Ideen? Wir sind damals davon ausgegangen, dass das ganze Texturengedöns nur in den Animationen vorkommt und haben das quasi dort reingeschoben. Jetzt wird es hier aber auch nochmal gebraucht? 

> 
> > @AHeinisch An welcher Stelle hattest Du Dir `Function<String, String>` vorgestellt? Als Parameter für `Image`? Dann würde ich als Nutzer den String nicht selbst ausrechnen, sondern die Berechnungsvorschrift übergeben, und es könnte für eine Handvoll von Use-Cases vordefinierte Funktionen geben? Hmmm 🤔
> 
> Also ich hatte es mir gedacht, um das Extrahieren des Dateipfades aus der Aufgabenstellung von außen konfigurierbar zu machen, anstelle eines fest codierten regex. Dies ist aber glaube eher was für ein neues Issue.

Ja.